drivers/modem/gsm_ppp.c

```diff
@@ -1107,16 +1114,49 @@ void gsm_ppp_start(const struct device *dev)
 	(void)gsm_work_reschedule(&gsm->gsm_configure_work, K_NO_WAIT);
 }
 
-void gsm_ppp_stop(const struct device *dev)
+void gsm_ppp_cancel(struct gsm_modem *gsm)
 {
-	struct gsm_modem *gsm = dev->data;
-	struct net_if *iface = gsm->iface;
 	struct k_work_sync work_sync;
 
 	(void)k_work_cancel_delayable_sync(&gsm->gsm_configure_work, &work_sync);
 	if (IS_ENABLED(CONFIG_GSM_MUX)) {
 		(void)k_work_cancel_delayable_sync(&gsm->rssi_work_handle, &work_sync);
 	}
+}
+
+void gsm_ppp_recover_cmux(const struct device *dev)
```

Comment:
Please remove the chip specific comments (if we plan on merging it back to zephyr repo)